### PR TITLE
Attempts to improve the "attachments" models.

### PR DESCRIPTION
**Pull request by gohrner**
There's lots of redundancy in the API spec which leads to generation of
duplicate but distinct / incompatible model classes during code
generation.

The "attachment" object type is only one of several places where this
occurs in the spec, but I'd get feedback if this unification is desired
first before investing more time in cleaning up definitions.

Please show some patience, as I'm not yet very experienced with OpenAPI V3.

See also https://community.openproject.org/topics/14323

### CODE REVIEW

File: docs/api/apiv3/components/schemas/attachments_model.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Schema: Attachments
+---
+type: object
+properties:
```
Comment (Kharonus) [inline docs/api/apiv3/components/schemas/attachments_model.yml]: Here, the same problem hits like in #10179, I'd like to get rid of those redundant schema declarations for collections step by step.

Can you cast another glance, how those classes are usable from your generated client?

**Review by Kharonus — CHANGES_REQUESTED** [review]
We need to solve that collection schema challenge.

File: docs/api/apiv3/components/schemas/attachments_model.yml (hunk 1 above)
Comment (gohrner) [reply docs/api/apiv3/components/schemas/attachments_model.yml]: Not quite sure what you mean - the `allOf` approach to integrate the `collection_model.yml`, which we're discussing in [PR 10179](https://github.com/opf/openproject/pull/10179#discussion_r830523691), or something else?

File: docs/api/apiv3/components/schemas/attachments_model.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        type: object
+        properties:
+          elements:
+            type: array
+            readOnly: true
+            items:
+                allOf:
+                - "$ref": "./attachment_model.yml"
+                - description: Collection of Attachments
+                readOnly: true
```
Comment (Kharonus) [inline docs/api/apiv3/components/schemas/attachments_model.yml]: Like in https://github.com/opf/openproject/pull/10179 please remove the `readOnly`from the items declaration. on property `elements` it is sufficient.

**Review by Kharonus — CHANGES_REQUESTED** [review]
single thing remaining

File: docs/api/apiv3/components/schemas/attachments_model.yml (hunk 1 above)
Comment (Kharonus) [reply docs/api/apiv3/components/schemas/attachments_model.yml]: Yes, sorry for being not explicit.

The current state looks fine, except this last small thing.

File: docs/api/apiv3/components/schemas/attachments_model.yml (hunk 2 above)
Comment (gohrner) [reply docs/api/apiv3/components/schemas/attachments_model.yml]: Ok, please have a look.

File: docs/api/apiv3/components/schemas/attachments_model.yml (hunk 1 above)
Comment (gohrner) [reply docs/api/apiv3/components/schemas/attachments_model.yml]: Just for confirmation:

If `users_model.yml` and `watchers_model.yml` are different and should not be merged, that's different with the models I merged here and it's fine to unify those?

Comment (Kharonus) [reply docs/api/apiv3/components/schemas/attachments_model.yml]: yes, those different attachment models were basically generated automatically, when we created the initial version of the spec from grape.

But there are no differences in the attachment models, we can unitfy them. Thanks for your effort.

**Review by Kharonus — COMMENTED** [review]
LGTM

**Review by Kharonus — APPROVED** [review]
LGTM

**Pull request merged**